doc/source/data/transforming-datasets.rst

```diff
@@ -78,15 +80,253 @@ express your customized business logic in transformations. Here we will focus on
+  in-memory conversions of Pandas DataFrame data to a ``Dataset`` (e.g. converting from
+  Pandas, Dask, Modin, Spark, and Mars) will use Pandas DataFrames as the internal data
+  representation in order to avoid extra data conversions/copies.
+
+* **Tensor Datasets:** Represented under-the-hood as a single-column
+  `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`__
+  or
+  `Pandas DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__,
+  using our :ref:`tensor extension type <dataset-tensor-extension-api>` to embed the
+  tensor in these tables under a single ``"__value__"`` column. Tensor datasets are
+  created by reading on-disk tensor data (NPY), converting from in-memory tensor data
+  (NumPy), and by returning tensor data (NumPy) from previous batch UDFs.
+
+  Note that converting the underlying tabular representation to a NumPy ndarray is
+  zero-copy, including converting the entire column to an ndarray and converting just
+  a single column element to an ndarray.
+
+* **Simple Datasets:** Represented under-the-hood as plain Python lists. Simple
+  datasets are created by reading on-disk binary and plain-text data, converting from
+  in-memory simple data (Python lists), and by returning simple data (Python lists)
+  from previou batch UDFs.
+
+  Simple datasets are mostly used as an escape hatch for data that's not cleanly
+  representable in Arrow Tables and Pandas DataFrames.
+
+Batch Formats
+=============
+
+Batch UDFs have the following batch format options, which control the format of the data
+batches that are passed to the provided batch UDF.
+
+.. tabbed:: "native" (default)
+
+  The ``"native"`` batch format presents batches in the canonical format for each dataset
+  type:
+
+  * **Tabular (Arrow or Pandas) Datasets:** Each batch will be a
+    `pandas.DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__.
+    If the dataset is represented as Arrow Tables under-th-ehood, the conversion to the
+    Pandas DataFrame batch format will incur a conversion cost (i.e., a full copy of
+    each batch will be created in order to convert it).
+  * **Tensor Datasets:** Each batch will be a
+    `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`__.
+  * **Simple Datasets:** Each batch will be a Python list.
+
+  **TODO:** Add example.
+
+.. tabbed:: "pandas"
+
+  The ``"pandas"`` batch format provides batches in a
+  `pandas.DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__
+  format. If converting a simple dataset to Pandas DataFrame batches, a single-column
+  dataframe with the column ``"value"`` will be created.
+
+  If the underlying datasets data is not already in Pandas DataFrame format, the
+  batch format conversion will incur a copy for each batch.
+
+  **TODO:** Add example.
+
+.. tabbed:: "pyarrow"
```

Comment:
Isn't it just "arrow"?

Reply:
The batch format is `"pyarrow"`, although I've always preferred `"arrow"`, and this would also align with the `"arrow"` dataset format.